### PR TITLE
[SPARK-12386] [core] Fix NPE when spark.executor.port is set.

### PR DESCRIPTION
**Pull request by vanzin**

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/SparkEnv.scala [hunk 1]
```diff
@@ -256,7 +256,14 @@ object SparkEnv extends Logging {
...
         rpcEnv.asInstanceOf[AkkaRpcEnv].actorSystem
       } else {
-        val actorSystemPort = if (port == 0) 0 else rpcEnv.address.port + 1
+        val actorSystemPort =
+          if (port == 0) {
+            0
+          } else if (rpcEnv.address != null) {
+            rpcEnv.address.port + 1
+          } else {
+            port + 1
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/SparkEnv.scala]: could you use `port` directly here since `rpcEnv` doesn't use it in this case?

**Comment by zsxwing** [comment]
LGTM pending tests

**Comment by markhamstra** [comment]
I don't know... it's longer than the one-liner I was told to expect. :)

**Comment by zsxwing** [comment]
retest this please

**Comment by vanzin** [comment]
>  it's longer than the one-liner

I deleted just one line!

**Comment by vanzin** [comment]
retest this please

File: core/src/main/scala/org/apache/spark/SparkEnv.scala [hunk 2]
```diff
@@ -256,7 +256,14 @@ object SparkEnv extends Logging {
       if (rpcEnv.isInstanceOf[AkkaRpcEnv]) {
         rpcEnv.asInstanceOf[AkkaRpcEnv].actorSystem
       } else {
-        val actorSystemPort = if (port == 0) 0 else rpcEnv.address.port + 1
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/SparkEnv.scala]: @markhamstra we can make this a real oneliner if we use Option :)
```
val actorSystemPort = if (port == 0) 0 else Option(rpcEnv.address).map(_.port + 1).getOrElse(port)
```
not sure if that's easier or harder to read. Either way is fine.

**Comment by andrewor14** [comment]
LGTM

File: core/src/main/scala/org/apache/spark/SparkEnv.scala (hunk 2 above)
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/SparkEnv.scala]: It's simpler: `val actorSystemPort = if (port == 0 || rpcEnv.address == null) port else port + 1`

**Comment by vanzin** [comment]
flakiness? retest this please

File: core/src/main/scala/org/apache/spark/SparkEnv.scala (hunk 2 above)
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/SparkEnv.scala]: +1 to @zsxwing

Comment (vanzin) [inline core/src/main/scala/org/apache/spark/SparkEnv.scala]: Still doesn't fit in one line, though. But it's simpler.

**Comment by andrewor14** [comment]
Merging into master 1.6 thanks!

**Pull request closed without merging**